plugins/redfish/fu-redfish-backend.c

```diff
@@ -224,6 +226,61 @@ fu_redfish_backend_check_wildcard_targets(FuRedfishBackend *self)
 	}
 }
 
+static void
+fu_redfish_backend_set_session_key(FuRedfishBackend *self, const gchar *session_key)
+{
+	g_free(self->session_key);
+	self->session_key = g_strdup(session_key);
+}
+
+static size_t
+fu_redfish_backend_headers_callback(char *ptr, size_t size, size_t nmemb, void *userdata)
+{
+	g_autofree gchar *session_key = NULL;
```

Comment:
can you move this into the `if` scope please? I like to keep everything as *local* as we can.